### ISSUE TITLE
Determine how to add tests

### ISSUE THREAD

**Issue by rasshai** [chiptiptech/connect#3]
As the project is getting bigger, we need tests to avoid degradation. Before we add tests, let's determine how to add tests.
- Which testing module should we use?
- Where do we put the test code?

There are four different types of components in our project. We have to decide how test for each component.
- Platform hardware (under connect/platform/hardware)
- Platform software (under connect/platform/software)
- Application hardware (under connect/apps/<app name>/hardware)
- Application software (under connect/platform/software/bin <- the directory for placing application software is another issue that has to be fixed!)

Maybe we can learn from another open-source python-based project how to do software testing. Hardware testing is quite straightforward.

**Comment by oika** [comment]
Did you diside not to use nose? (I saw #26. )

**Comment by rasshai** [comment]
Not really. I'm just trying it out. I also want to try pytest. I will decide which to use before I create first version of all the tests.

**Comment by rasshai** [comment]
I'm doing some research on unit tests for python, and finding many articles that says pytest is better than nose2 or unittest. I will probably use pytest after trying it. https://www.reddit.com/r/Python/comments/3ys29v/nose_alternatives_nose2_pytest_or_something_else/
